metadata-ingestion/src/datahub/ingestion/source/schema_inference/csv_tsv_jsonl.py

```diff
@@ -72,3 +72,13 @@ def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
         # infer schema of a tsv file without reading the whole file
         table = Table(file, format="tsv")
         return get_table_schema_fields(table, max_rows=self.max_rows)
+
+
+class JsonlInferrer(SchemaInferenceBase):
+    def __init__(self, max_rows: int):
+        self.max_rows = max_rows
+
+    def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+        # infer schema of a jsonl file without reading the whole file
+        table = Table(file, format="jsonl")
+        return get_table_schema_fields(table, max_rows=self.max_rows)
```

Comment:
the get_table_schema_fields method is specialized for neat tables, and I don't think it handles nested objects well

I'd rather use the same mechanism as json's schema inference, and just use existing `jsl.Reader` with a limit to implement max_rows. We can add a flag to `JsonInferrer` to assume jsonl and skip trying a normal, full load of the json

That means we can undo the rename of csv_tsv.py, and put that stuff in the json.py file instead

Reply:
@hsheth2 This makes sense. Let me make the changes for this.

Reply:
Can you please re-review the changes? Thanks!